src/Functions/helpers.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+namespace mheap\GithubActionsReporter\Functions;
+
+use mheap\GithubActionsReporter\Printer6;
+use mheap\GithubActionsReporter\Printer7;
+use mheap\GithubActionsReporter\Printer8;
+use mheap\GithubActionsReporter\Printer9;
+use PHPUnit\Framework\TestFailure;
+use ReflectionClass;
+
+/**
+ * @param $version
+ *
+ * @return string|null Fully Qualified Class Name, or null if no matching version
+ *
+ * @internal
+ */
+function determinePrinter($version)
+{
+    $versionMatrix = [
+        // greater than equals, lower than equals, printer FQCN
+        ['6.0', '6.99.99', Printer6::class],
+        ['7.0', '7.99.99', Printer7::class],
+        ['8.0', '8.99.99', Printer8::class],
+        ['9.0', true, Printer9::class],
+    ];
+
+    foreach ($versionMatrix as list($lowerVersion, $upperVersion, $class)) {
+        if (
+            version_compare($version, $lowerVersion, '>=') == true &&
+            ($upperVersion === true || version_compare($version, $upperVersion, '<=') == true)
+        ) {
+            return $class;
+        }
+    }
+
+    return null;
+}
+
+/**
+ * @param TestFailure $defect
+ * @param string $defectType
+ *
+ * @return string
+ * @throws \ReflectionException
+ * @internal
+ */
+function printDefectTrace($defect, $defectType)
+{
+    $e = $defect->thrownException();
+
+    $errorLines = array_filter(
+        explode("\n", (string)$e),
+        static function ($l) {
+            return $l;
+        }
+    );
+
+    $error = end($errorLines);
+    $lineIndex = strrpos($error, ":");
+    $path = substr($error, 0, $lineIndex);
+    $line = substr($error, $lineIndex + 1);
+
+    list($reflectedPath, $reflectedLine) = getReflectionFromTest(
+        $defect->getTestName()
+    );
+
+    if ($path !== $reflectedPath) {
+        $path = $reflectedPath;
+        $line = $reflectedLine;
+    }
+
+    $message = explode("\n", $defect->getExceptionAsString());
+    $message = implode('%0A', $message);
+
+    // Some messages might contain paths. Let's convert thost to relative paths too
+    $message = relativePath($message);
+    $lineFeedPosition = strpos($message, '%0A');
+    if (is_int($lineFeedPosition) === true) {
```

Comment:
This is an incorrect change imo. The old behaviour stripped linefeeds from the end of the error, whilst the new behaviour truncates any error messages that contain a line feed. Could we revert to the old behaviour?

Reply:
that's my bad - I wasn't 100% sure if it was intentional or not, I'll revert it and leave a comment to avoid future developer confusion

Reply:
@mheap I'm not sure why, but reverting it is not playing ball with PHPUnit 6
see https://github.com/VeryStrongFingers/phpunit-github-actions-printer/runs/1377052224?check_suite_focus=true#step:8:20
any ideas?

Reply:
This is very strange. It looks as though PHPUnit 6 does not add the additional information.

We could have a `.phpt` file just for PHPUnit 6, or update the printer to add this information (we have it available in `file` and `line`)